torch/fx/experimental/Partitioner.py

```diff
@@ -180,6 +335,101 @@ def create_partition(self) -> Partition:
         """Create a partition and append it to self.partitions."""
         partition_id = len(self.partitions)
         assert isinstance(self.graph_module, GraphModule)
-        partition = Partition(partition_id, self.graph_module)
+        partition = Partition(partition_id, self.graph_module, self.orphans)
         self.partitions.append(partition)
         return partition
+
+    def combine_partitions_based_on_size(self, available_mem) -> None:
+        """Combining small partitions together to keep as less partitions as possible.
+           Here is an example of the algorithm to do this:
+           Assume some partitions, we first sort them based on partiiton used memory size.
+           [(partition_4, 1), (partition_3, 1), (partition_2, 2), (partition_1, 7), (partition_0, 9)]
+           The available memory is 10.
+
+           step 1: self.find_partition_to_combine_based_on_size()
+           First, look the smallest partition, partition_4: 10 - 1 = 9
+           It means any partition has a used memory equal or less than 9 could combine this partition
+           We go from the largest and selection partition_0.
+           As a result, partition_9 has a size of 10
+
+           Then, repeat step 1, for partition_3, 10 - 1 = 9
+           partition_1 has a size of 7 and could combine with partition_3
+           For partition_2, 10 - 2 = 8
+           At this moment, partition_1 has a size of 8 and it combines partition_2
+        """
+        flag = True
+        while flag:
+            sorted_partitions = sorted(self.partition_to_used_mem.items(), key=lambda item: item[1])
+            smallest_partition = sorted_partitions[0][0]
+            left_mem = available_mem - self.partition_to_used_mem[smallest_partition]
+            flag = self.find_partition_to_combine_based_on_size(sorted_partitions, left_mem)
+        return
+
+    def find_partition_to_combine_based_on_size(self, sorted_partitions: List[Tuple[Partition, int]], left_mem: int) -> bool:
+        """step 1 in self.combine_partition_based_on_size()"""
+        flag = False
```

Comment:
nit: We already know it's a flag since it's a boolean :) Can you name this something more descriptive?